hazelcast/test/src/HazelcastTests7.cpp

```diff
@@ -1845,6 +1845,27 @@ TEST_F(cloud_discovery_test, invalid_token)
                  exception::illegal_state);
 }
 
+TEST_F(cloud_discovery_test, token_should_not_be_leaked)
+{
+    auto config = get_config();
+    config.get_connection_strategy_config()
+      .get_retry_config()
+      .set_cluster_connect_timeout(std::chrono::milliseconds(100));
+    auto& cloudConfig = config.get_network_config().get_cloud_config();
+    cloudConfig.enabled = true;
+    cloudConfig.discovery_token = "bwMqx9xp4RBkBWfdpE3WPsuCLd0CXNS";
+
+    auto discovery_token = cloudConfig.discovery_token;
+
+    try {
+        hazelcast::new_client(std::move(config)).get();
+        FAIL();
+    } catch (const exception::illegal_state& e) {
+        std::string message = e.what();
+        ASSERT_EQ(message.find(discovery_token), std::string::npos);
```

Comment:
you should also check the nested exception message!!!

Reply:
Nested exception contains it.

Reply:
Changed the implementation, `token` is filtered out in the class which is the source of the exception. So it is not leaked by neither inner nor outer exceptions.

There is a problem related with an unit test, I wrote the test to check inner exception message but `std::rethrow_if_nested` doesn't throw the inner exception. Added it to comment.

Reply:
Let's öpen an issue for this.

Reply:
I will open after merge with the link of the test.